byoqm/qualitymodel/treequalitymodel.py

```diff
@@ -0,0 +1,50 @@
+from abc import ABC, abstractmethod
+from ast import List
+from typing import Callable
+from byoqm.metric.metric import Metric
+from qualitymodel import Characteristic, QualityModel
+
+class Node(Characteristic):
+    def __init__(self, name: str, parent: Node, weights: List[int | float]):
+        self.name = name
+        self.parent = parent
+        self.children = []
+        self.weights = weights
+
+    def measure(self):
+        measurement = None
+        for c in self.children:
+            measurement += c.measure()
+        return measurement
+
+
+class TreeQualityModel(QualityModel):
+    def __init__(self):
+        self.root = Node("root", None, [])
```

Comment:
IMO making the root's name "quality" makes more sense. What are people's thoughts?

Reply:
Could do that yeah, that would go nicely with our overall abstraction and drawing of the Quality Model Tree.

Reply:
Will change it.